### DISCUSSION TITLE
Allow "pinning" of removable devices

### DISCUSSION THREAD

**Discussion by pentamassiv** [fwupd/fwupd#4958] [Ideas]
**Is your feature request related to a problem? Please describe.**
Some devices are removable. They only get listed if they are connected but I also want to know if there is an update available for a disconnected device without having to connect it.

**Describe the solution you'd like**
With the implementation of https://github.com/fwupd/fwupd/commit/519e80b1eeb89670031ddf415c02bf618c52ea25, it is possible to list devices that can't be updated right now because of a user fixable problem. The device being disconnected is a user fixable problem, but there need to be a way for fwupdmgr to know about these disconnected devices. It would be nice if you could "pin" a connected device. This could work as follows:
- fwupdmgr get-devices
- fwupdmgr pin [DEVICE-ID|GUID]
- disconnect the removable device
- fwupdmgr get-updates
  --> The pinned device is still listed if there are updates but it is noted that the device has to be connected to be updated

Of course there would also need to be a "unpin" command to get rid of a pinned device.

**Describe alternatives you've considered**
Connect the device each time you check for firmware updates

**Comment by superm1** [comment]
To me this sounds like it could be solved by certain devices being replayed from the history database but as disconnected. It's then very similar to devices that are ble but out of range.

At least conceptually it makes sense to me. There was a previous request for something like this but we didn't have some of that infrastructure available. It seems more achievable now.
